clients/src/main/java/org/apache/kafka/common/requests/RequestHeader.java

```diff
@@ -16,34 +16,36 @@
  */
 package org.apache.kafka.common.requests;
 
-import static org.apache.kafka.common.protocol.Protocol.REQUEST_HEADER;
-
-import java.nio.ByteBuffer;
-
 import org.apache.kafka.common.protocol.Protocol;
-import org.apache.kafka.common.protocol.types.Field;
+import org.apache.kafka.common.protocol.types.Schema;
 import org.apache.kafka.common.protocol.types.Struct;
 
+import java.nio.ByteBuffer;
+
 /**
  * The header for a request in the Kafka protocol
  */
 public class RequestHeader extends AbstractRequestResponse {
-
-    private static final Field API_KEY_FIELD = REQUEST_HEADER.get("api_key");
-    private static final Field API_VERSION_FIELD = REQUEST_HEADER.get("api_version");
-    private static final Field CLIENT_ID_FIELD = REQUEST_HEADER.get("client_id");
-    private static final Field CORRELATION_ID_FIELD = REQUEST_HEADER.get("correlation_id");
+    private static final String API_KEY_FIELD_NAME = "api_key";
+    private static final String API_VERSION_FIELD_NAME = "api_version";
+    private static final String CLIENT_ID_FIELD_NAME = "client_id";
+    private static final String CORRELATION_ID_FIELD_NAME = "correlation_id";
 
     private final short apiKey;
     private final short apiVersion;
     private final String clientId;
     private final int correlationId;
 
     public RequestHeader(Struct struct) {
-        apiKey = struct.getShort(API_KEY_FIELD);
-        apiVersion = struct.getShort(API_VERSION_FIELD);
-        clientId = struct.getString(CLIENT_ID_FIELD);
-        correlationId = struct.getInt(CORRELATION_ID_FIELD);
+        apiKey = struct.getShort(API_KEY_FIELD_NAME);
+        apiVersion = struct.getShort(API_VERSION_FIELD_NAME);
+
+        // only v0 of the controlled shutdown request is missing the clientId
+        if (struct.hasField(CLIENT_ID_FIELD_NAME))
+            clientId = struct.getString(CLIENT_ID_FIELD_NAME);
+        else
+            clientId = null;
```

Comment:
Do we want to set this to the empty string so that it's consistent with the case where it's unset?